MessageStackView/MessageManager.swift

```diff
@@ -173,59 +239,144 @@ open class MessageManager
+    /// When a `UIView` wants to be dismissed on tap,
+    @objc func viewTapped(_ sender: UITapGestureRecognizer) {
+        guard sender.state == .ended, let view = sender.view else {
+            return
+        }
+        remove(view: view, animated: true)
     }
     
     // MARK: - Invalidate
     
     /// Invalidate and remove timers
     public func invalidate() {
+        messageStackView.arrangedSubviewsExcludingSpace.forEach {
+            remove(view: $0, animated: false)
+            removeTapGesture(from: $0)
+        }
+
+        // These should theoretically be cleaned up from the remove methods,
+        // but add in case of unexpected manipulation from caller
         timerMap.values.forEach {
             $0.invalidate()
         }
         timerMap = [UIView: Timer]()
+        tapGestureMap = [UIView : UITapGestureRecognizer]()
     }
+}
+
+// MARK: - Extensions
+
+extension MessageDismiss {
+    
+    /// Default `MessageDismiss`
+    public static let `default`: MessageDismiss = .after(3)
+}
+
+extension PostAnimation {
     
-    // MARK: - Private API
+    /// Default `PostAnimation`
+    public static let `default`: PostAnimation = .both
+}
+
+extension MessageView {
     
-    /// Convert `message` model to `messageView`
-    private func configure(messageView: MessageView, for message: Message) {
+    /// Apply `Message` to `MessageView`
+    func apply(message: Message) {
+        
         // Title
-        messageView.titleLabel.text = message.title
+        titleLabel.text = message.title
         
         // Detail
-        let subtitle = message.subtitle?.trim ?? ""
-        messageView.subtitleLabel.text = subtitle
-        messageView.subtitleLabel.isHidden = subtitle.isEmpty
+        subtitleLabel.setTextAndHidden(message.subtitle)
         
-        // Image
-        messageView.imageView.image = message.image
-        messageView.imageView.isHidden = message.image == nil
+        // Left Image
+        leftImageView.setImageAndHidden(message.leftImage)
         
-        // `MessageConfiguration`
-        messageView.apply(configuration: messageConfiguation)
+        // Right Image
+        rightImageView.setImageAndHidden(message.rightImage)
     }
 }
 
-
-// MARK: - Extensions
-
-extension MessageDismiss
-{
-    /// Default `MessageDismiss`
-    public static let `default`: MessageDismiss = .after(3)
+extension UIImageView {
+    
+    /// Set the `UIImageView.image` and update `isHidden` by nullability of`image`
+    func setImageAndHidden(_ image: UIImage?) {
+        self.image = image
+        isHidden = image == nil
+    }
+    
 }
 
-extension PostAnimation
-{
-    /// Default `PostAnimation`
-    public static let `default`: PostAnimation = .both
+extension UILabel {
+    
+    /// Set the `UILabel.text` and update `isHidden` by nullability of `text`
+    func setTextAndHidden(_ text: String?) {
+        let labelText = text ?? ""
+        self.text = text
+        isHidden = labelText.isEmpty
```

Comment:
Normally to be super safe for this stuff I'll do:
`labelText.trimmingCharacters(in: .whitespacesAndNewLines).isEmpty` :D